modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2844,6 +2844,14 @@ public CodegenProperty fromProperty(String name, Schema p) {
 
         if (p.getDeprecated() != null) {
             property.deprecated = p.getDeprecated();
+        } else if (p.get$ref() != null) {
+            String ref = ModelUtils.getSimpleRef(p.get$ref());
+            if (ref != null) {
+                Schema referencedSchema = ModelUtils.getSchemas(this.openAPI).get(ref);
+                if (referencedSchema != null && referencedSchema.getDeprecated() != null) {
+                    property.deprecated = referencedSchema.getDeprecated();
+                }
+            }
```

Comment:
@asmundg thanks for the PR. In https://github.com/OpenAPITools/openapi-generator/pull/6093/files#diff-7cb46fa53f89a458a7b7cb201d2214a8R2820, we should have "unalias" (dereference) the schema so looks like something wrong with the unaliasing that we need to fix. I'll take another look when I've time later this week.

Reply:
Gah, I lost a comment block in rebasing there. Thanks for pointing it out.